### ISSUE TITLE
External auth in 2016.11.9 fails on attempt to resolve group

### ISSUE THREAD

**Issue by gordonm** [saltstack/salt#46626]
### Description of Issue/Question
External auth in`2016.11.9` release seems to have been broken by #45130.

### Setup
Our external auth configuration includes the following: 
```
external_auth:
  pam:
    <group>%:
      - '@runner'
      - 'G@role:<role>':
        - test.ping
        - grains.*
```
We use external authentication to allow certain groups to run orchestration tasks on the master through the Salt API, this allows them to read the job output using the API.

### Steps to Reproduce Issue
Attempts to authenticate to the Salt API using external auth result in the following exception on the master:
```
2018-03-19 18:31:01,035 [salt.transport.zeromq][ERROR   ][9065] Some exception handling a payload from minion
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/salt/transport/zeromq.py", line 620, in handle_message
    ret, req_opts = yield self.payload_handler(payload)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 870, in run
    value = future.result()
  File "/usr/lib/python2.7/dist-packages/tornado/concurrent.py", line 215, in result
    raise_exc_info(self._exc_info)
  File "/usr/lib/python2.7/dist-packages/tornado/gen.py", line 215, in wrapper
    result = func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 897, in _handle_payload
    'clear': self._handle_clear}[key](load)
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 911, in _handle_clear
    return getattr(self.clear_funcs, load['cmd'])(load), {'fun': 'send_clear'}
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 1802, in runner
    'runner')
  File "/usr/lib/python2.7/dist-packages/salt/master.py", line 1724, in process_token
    groups = token.get['groups']
TypeError: 'builtin_function_or_method' object has no attribute '__getitem__'
```

### Versions Report
```
Salt Version:
           Salt: 2016.11.9

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: 0.26.0
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.3
      pycparser: 2.18
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: 0.26.3
         Python: 2.7.6 (default, Nov 23 2017, 15:49:48)
   python-gnupg: 0.3.6
         PyYAML: 3.10
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-132-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

### Workaround
We restored service by replacing `/usr/lib/python2.7/dist-packages/salt/master.py` with https://raw.githubusercontent.com/saltstack/salt/5f90812b122bf8b78ddbf94e99cccdbd4cc19203/salt/master.py, which reverts the commit from #45130.

**Comment by Ch3LL** [comment]
I'm guessing if you were to change `groups = token.get['groups']` to `groups = token.get('groups')` it should work. 

I was looking into if this was the same in 2017.7 and I don't see the code being merged forward assuming because of a merge conflict here: https://github.com/saltstack/salt/pull/45203

ping @rallytime is there are reason this didn't make it in the merge forward or maybe i'm misunderstanding something.

**Comment by mgzabriskie** [comment]
I will second this. We are having this issue in 2016.11.9 and 2016.11.10. @Ch3LL 's suggested change is a valid workaround. I would love to see this get patched for real though.

**Comment by Ch3LL** [comment]
unfortunately 2016.11 is a CVE security release only now so there are no longer any bug fixes being submitted to this branch. 2017.7 does seem to have this fix though which is a bug release fix currently.

**Comment by Ch3LL** [comment]
since this is resolved in 2017.7 i will go ahead and close here, but please let me know if its still not working in 2017.7
